app/api/discount_codes.py

```diff
@@ -23,14 +23,27 @@ class Meta:
         self_view_kwargs = {'id': '<id>'}
         inflect = dasherize
 
-    @validates_schema
-    def validate_quantity(self, data):
-        if 'max_quantity' in data and 'min_quantity' in data:
-            if data['max_quantity'] < data['min_quantity']:
+    @validates_schema(pass_original=True)
+    def validate_quantity(self, data, original_data):
+        original_data_attributes = original_data['data']['attributes']
+        discount_code = DiscountCode.query.filter_by(id=original_data['data']['id']).one()
+        if 'min_quantity' in original_data_attributes:
```

Comment:
@shubham-padia @poush 
>Also if ends_at is not present in patch, instead of this whole long process, just fetch end_date in before_update_object and add it to the data

Still that needs to be done in the `before_update_object` ( as it had been done earlier ) and then validation would take place in `before_update_object` instead of in `validate_date`.

It was changed on your suggestion in the first review.

I think it will be better if we have all the validation in the validating function itself.